megamek/src/megamek/common/Terrain.java

```diff
@@ -386,29 +388,52 @@ public int movementCost(Entity e) {
                 } else {
                     mp = 1;
                 }
+
+                if (isCrossCountry) {
```

Comment:
Sorry to chime in. 
This only checks the movement mode. WHEELED is also used by QuadVees in wheeled mode. A vehicle pilot probably shouldn't command a qv but MM doesn't prevent this. I wonder if e.isVehicle() should be added to the crossCountry boolean definiton.

Also, I agree with Windchild that repeating mp *= 2 really calls for OR-ing the two conditions or it looks like an error.

Reply:
The first one for woods was a bug that was fixed. I can probably try to do the if/else or if/else if better, but to cram it all in a single if would make it look rather confusing imo.

As for quadvee. A check for isVee will get added just to make sure it ignores any weirdness like that.

Reply:
I agree these are usually ugly to read with all the ((()()((()((()))). But having the same result multiple times screams error or legacy issue. Maybe a solution is to add a 
final boolean allowedHoverWheeled = ((moveMode == EntityMovementMode.HOVER) || (moveMode == EntityMovementMode.TRACKED)) && (level == 6);
and then if (allowedHoverWheeled || (moveMode == EntityMovementMode.WHEELED)) {
This is still shorter than the if else and readable enough I guess. What do you say?

Reply:
Eh hoverTracked of course

Reply:
It could work, I will split the changes into multiple commits for review to make it easier to revert if it doesn't look right and you decide against it. Quadvee will be first then the others.